didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+    # message needs to be forwarded, the RoutingService will handle wrapping
+    # each encrypted message within a forward message. The built-in
+    # RoutingService allows for multiple levels of relays, so you don't need to
+    # worry about how a message is routed to the recipient.
+    router = RoutingService(packaging=packer, resolver=resolver)
+
+    # Once everything is setup, we need to store the DID Secrets within the
+    # SecretsService. We do this by taking the secrets that were passed in,
+    # converting them to AskarSecretKey objects (since we are using the
+    # AskarCryptoService), then we associate them with the public key lookup
+    # value that will be sent to us.
+    verkey, xkey = did_secrets
+    await secrets.add_secret(AskarSecretKey(verkey, f"{did}#key-1"))
+    await secrets.add_secret(AskarSecretKey(xkey, f"{did}#key-2"))
+
+    # These can be removed once all keys being sent/received are in #key-N
+    # format. They basically do the same as above, except with the same public
+    # keys that are returned from the CompatibilityPrefixResolver.
+    doc = await resolver.resolve_and_parse(did)
+    await secrets.add_secret(AskarSecretKey(verkey, doc.authentication[0]))
+    await secrets.add_secret(AskarSecretKey(xkey, doc.key_agreement[0]))
+
+    # Finally, we put it all together in the DIDCommMessaging class. The
+    # DIDCommMessaging handles the orchestration of each individual service,
+    # ensuring that messages get packed and delivered via a simple and straight
+    # forward interface.
+    DMP = DIDCommMessaging(
+        crypto=crypto,
+        secrets=secrets,
+        resolver=resolver,
+        packaging=packer,
+        routing=router,
+    )
+
+    return DMP
+
+
+async def send_http_message(
+    dmp: DIDCommMessaging, my_did: DID, message: Message, target: DID
+):
+    """Send a message via HTTP."""
+
+    # Get the message as a dictionary
+    message_wrapper = message
+    message = message_wrapper.as_dict()
+
+    # Encrypt/pack the message to our target
+    packy = await dmp.pack(
+        message=message,
+        to=target,
+        frm=my_did,
+    )
+    packed = packy.message
+
+    # Get the first http endpoint from the last DID in the DID chain
+    endpoint = packy.get_endpoint("http")
+
+    async with aiohttp.ClientSession() as session:
+        LOG.info("posting message type %s to %s" % (message_wrapper.type, endpoint))
+
+        async with session.post(endpoint, data=packed) as resp:
+            LOG.debug("posted message: %s" % (message))
+            LOG.debug("message ID: %s" % (message_wrapper.id))
+            packed = await resp.text()
+
+            # If the HTTP enpoint responded with a message, decode it
+            if len(packed) > 0:
+                unpacked = await dmp.packaging.unpack(packed)
+                msg = unpacked[0].decode()
+                LOG.debug("Raw message from remote %s" % msg)
+                return Message.from_json(msg)
```

Comment:
This needs error handling for non-200 status codes and should probably account for the possibility of receiving a response that is not a packed message (like the plaintext didcomm messages the mediator may send when there are issues).

Reply:
We are now raising an exception if it is non-200, detailing the error